### PR TITLE
Prepare refactoring of notifications (removal of NamedInts)

### PR DESCRIPTION
**Pull request by rloutrel**
#2273 : First 2 tests + add typing

Sorry, I could not understand how to use mock.Mock.

This could be merged already or wait for actual re-factoring that I will try to do next week.

### CODE REVIEW

**Comment by pfps** [comment]
@MattHag Can you review this PR?   I invited you into the project at a level that I think allows you to be a suggested reviewer.

**Comment by MattHag** [comment]
Yes

**Comment by pfps** [comment]
@MattHag OK, I set you up as a reviewer.

File: tests/logitech_receiver/test_notifications.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        (Registers.DISCOVERY_STATUS_NOTIFICATION, b"\x01", "device_timeout", None),
+        (Registers.PAIRING_STATUS_NOTIFICATION, b"\x02", "failed", None),
+    ],
+)
+def test_process_receiver_notification_bolt_pairing_error(sub_id, notification_data, expected_error, expected_new_device):
+    low_level_mock = mock.Mock(spec=LowLevelInterface)
+    serial_reply = b"\x00\x00\x00\x00\x00\x00\x00\x00"
+    low_level_mock.read_register = mock.Mock(return_value=serial_reply)
+
+    receiver: Receiver = Receiver(MockLowLevelInterface(), None, {}, True, None, None)
```
[the remark this answers is not in the document]
Comment (MattHag) [reply tests/logitech_receiver/test_notifications.py:40]: Nevermind, it doesn't work with Mock alone.

**Review by MattHag — COMMENTED** [review]
Looks good, but I have clean up a littl a bit. Can you enable this option
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork
Then I can merge the change.

**Comment by rloutrel** [comment]
> Looks good, but I have done some clean up to merge it right away.
> 
> Please enable this and I can merge it, afterwards https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

It should already be activated. Is reviewer == maintainer ?

Did you already made the changes or shall I do it?

**Comment by rloutrel** [comment]
Ok I will tell you if my computer starts smoking because of you ;)

**Comment by MattHag** [comment]
Seems like I still can't push it. Here is the updated version
https://github.com/MattHag/Solaar/tree/rloutrel/master

**Comment by MattHag** [comment]
Yes, if you don't mind.

**Comment by rloutrel** [comment]
Now managed in https://github.com/MattHag/Solaar/tree/rloutrel/master

**Pull request closed without merging**